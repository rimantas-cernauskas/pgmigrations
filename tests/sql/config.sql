-- Test function pgmigrations.config()
\i tests/sql/setup/init.sql

BEGIN;
SELECT plan(8);

SET ROLE __pgmigrations_superuser;
SELECT is(
    pgmigrations.config('foo','bar'),
    TRUE,
    'superuser should be able to create new config entry'
);
SELECT is(
    pgmigrations.config('foo'),
    'bar',
    'superuser should be able to read existing config entry'
);
SELECT is(
    pgmigrations.config('foo','baz'),
    TRUE,
    'superuser should be able to update existing config entry'
);
SELECT is(
    pgmigrations.config('foo'),
    'baz',
    'value should be updated'
);
-- exception thrown when reading non existing entry
PREPARE my_query AS SELECT pgmigrations.config('bar');
SELECT throws_ok(
    'my_query',
    'Missing configuration for "bar"',
    'should throw exception when accessing non existing value'
);
DEALLOCATE my_query;

SET ROLE __pgmigrations_superuser;
SELECT is(
    pgmigrations.config('foo','bar'),
    TRUE,
    'setup - add expected configuration entry'
);
SET ROLE __pgmigrations_testuser;

SELECT is(
    pgmigrations.config('foo'),
    'bar',
    'unprivileged user should be able to read existing config entry'
);

PREPARE my_query AS SELECT pgmigrations.config('foo','bar');
SELECT throws_ok(
    'my_query',
    'permission denied for relation config',
    'should throw exception when attempting to create config option as unprivileged user');
DEALLOCATE my_query;

SELECT * FROM finish();
ROLLBACK;
