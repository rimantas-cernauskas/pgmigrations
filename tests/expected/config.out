-- Test function pgmigrations.config()
\i tests/sql/setup/init.sql
\set ECHO errors
1..8
ok 1 - superuser should be able to create new config entry
ok 2 - superuser should be able to read existing config entry
ok 3 - superuser should be able to update existing config entry
ok 4 - value should be updated
ok 5 - should throw exception when accessing non existing value
ok 6 - setup - add expected configuration entry
ok 7 - unprivileged user should be able to read existing config entry
ok 8 - should throw exception when attempting to create config option as unprivileged user
